fhirr4/ballerina/src/main/resources/fhirservice/fhir_preprocessor.bal

```diff
@@ -258,6 +560,16 @@ public isolated class FHIRPreprocessor {
 
         return processedSearchParams;
     }
+
+    isolated function handleSmartSecurity(r4:FHIRSecurity fhirSecurity, string? id) returns r4:FHIRError? {
+        r4:AuthzConfig? authzConfig = self.apiConfig.authzConfig;
+        if authzConfig is r4:AuthzConfig {
+            r4:FHIRError? handleSmartSecurityResult = r4:handleSmartSecurity(authzConfig, fhirSecurity, id);
+            if handleSmartSecurityResult is r4:FHIRError {
+                return handleSmartSecurityResult;
```

Comment:
Shall we capture a error log as well?

Reply:
What kind of info do you think we need to capture in the logs? Wouldn't the response we send to the client itself is descriptive enough to troubleshoot an issue?